src/main/kotlin/com/kryszak/gwatlin/http/BaseHttpClient.kt

```diff
@@ -54,11 +55,19 @@ internal open class BaseHttpClient(
         return processResult(result, ErrorResponse(response, RetrieveError::class.java))
     }
 
-    protected fun addDefaultHeaders(request: Request, language: String?) {
+    protected fun addDefaultHeaders(request: Request, language: ApiLanguage?) {
         schemaVersion?.let { request.appendHeader("X-Schema-Version" to it) }
-        request.appendHeader(Headers.ACCEPT_LANGUAGE to (language ?: defaultLanguage))
+
+        // Language priority list:
+        // 1. Language specified for this request, if any
+        // 2. Set default language for this client, if any
+        // If one of the mentioned languages is not null, the Accept-Language header gets set
+        arrayOf(language, defaultLanguage)
```

Comment:
If I read docs correctly, we do not need to send header for English locale, it's set by default if no locale is present on request. With that we can get rid of `defaultLanguage` field in class and simplify this logic to just 
```
language?.let {
    request.appendHeader(Headers.ACCEPT_LANGUAGE to it.apiString)
}
```

Reply:
End-user API clients do not allow to be created with specific language, so we should stick to passing desired language in methods only, or rework clients to accept language in constructor and then use it.

For me both solutions are fine.

Reply:
My idea with the `defaultLanguage` parameter was that you can just specify a language that should be used when constructing the API client. We can remove this option of course, since it doesn't seem necessary I guess.

Reply:
The idea is not bad, but as for now there are 0 clients with constructor that allows to create them with default language :smile: So as for now I'd say it would be best to get rid of it to simplify generic client code, we can later on change the way clients work and allow to create them. But for this change I'd say there are enough changes without that.